drivers/wifi/siwx91x/Kconfig.siwx91x

```diff
@@ -0,0 +1,48 @@
+# Copyright (c) 2024 Silicon Laboratories Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+config WIFI_SILABS_SIWX91X
+	bool "Silabs SiWx91x SoC series WiFi driver"
+	default y
+	depends on DT_HAS_SILABS_SIWX91X_WIFI_ENABLED
+	depends on NETWORKING
+	select WISECONNECT_NETWORK_STACK
+	select EVENTS
+	select NET_L2_WIFI_MGMT
+	help
+	  Enable WiFi driver for the Silabs SiWx91x SoC series.
+
+if WIFI_SILABS_SIWX91X
+
+choice
+	prompt "Network stack"
+	default WIFI_SILABS_SIWX91X_NET_STACK_NATIVE
+	help
+	  Choose "Native" stack if you want a better compatibility with Zephyr
+	  features. "Offloaded" stack may provide better resource (power and
+	  memory) consumption.
+
+config WIFI_SILABS_SIWX91X_NET_STACK_NATIVE
+	bool "Native"
+	select WIFI_USE_NATIVE_NETWORKING
+	select NET_L2_ETHERNET
+
+config WIFI_SILABS_SIWX91X_NET_STACK_OFFLOAD
+	bool "Offloaded"
+	select WIFI_OFFLOAD
+
+endchoice
+
+config NET_TCP_WORKQ_STACK_SIZE
+	default 2048
+
+config NET_RX_STACK_SIZE
+	default 2048
+
+config NET_MGMT_EVENT_STACK_SIZE
+	default 2048
+
+config NET_MGMT_EVENT_QUEUE_SIZE
+	default 10
```

Comment:
these do not apply anything if someone just enables the Kconfig, they should not be here

Reply:
Yeah, I am neither super fond of these lines. However, it seems to be a common pattern: `drivers/wifi/esp32/Kconfig.esp32`, `drivers/wifi/nrf_wifi/Kconfig.nrfwifi`, `arch/sparc/Kconfig`... I have tested several ways to set these values and I have not found any good way to declare them. Any idea?

Reply:
Not sure if these are just recommended values or minimums for it to work, @jukkar @krish2718 if they are minimums then there could be hidden Kconfigs added that specify the minimum e.g.
```
config NET_MGMT_EVENT_MINIMUM_STACK_SIZE
        int
        default 2048 if WIFI
        default 0

config NET_MGMT_EVENT_STACK_SIZE
        int
        range NET_MGMT_EVENT_MINIMUM_STACK_SIZE 32768
```

Reply:
I assume we can consider this is out of the scope of this PR.

Reply:
Yes, out of scope